### PR TITLE
#16132 Fix "Words that start with underscore are treated as keywords"

### LINKED ISSUE #16132: Words that start with underscore are treated as keywords
#### System information: 
- Operating system (distribution) and version: Fedora 35
- DBeaver version: Version 22.0.2.202204041542
- Additional extensions: N/A

#### Connection specification:
- Database name and version: Postgresql 12, 13
- Driver name: Postgres
- Do you use tunnels or proxies (SSH, SOCKS, etc)? NO

#### Describe the problem you're observing:
Keywords, when prefixed with non-literal characters are still highlighted as keywords. 
While a lot of those non-literals are not valid as starting characters for variables, underscore is an accepted one.
This results in highlighting words like _sql, _default as to be keywords. Which is obviously not true.
See the image below.
it's just a screenshot of a fragment of the code. Note % as variable name prefix will not work likely. Just to show behavior pattern.
image

#### Steps to reproduce, if exist:
See above

#### Include any warning/errors/backtraces from the logs
N/A

### PR DESCRIPTION
**Pull request by PILINING**

### CODE REVIEW

**Comment by PILINING** [comment]
#16132

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/AbstractSQLDialect.java [hunk 1]
```diff
@@ -316,7 +316,7 @@ public String[] getInnerBlockPrefixes() {
 
     @Override
     public boolean isWordStart(int ch) {
-        return Character.isUnicodeIdentifierStart(ch);
+    	return Character.isUnicodeIdentifierStart(ch) || ch == 95;
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/AbstractSQLDialect.java]:
```suggestion
        return Character.isUnicodeIdentifierStart(ch) || ch == '_';
```

**Review by E1izabeth — CHANGES_REQUESTED** [review]
HI, @PILINING! Thank you for contribution! Could you, please, make mentioned changes to make it clearer?

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/AbstractSQLDialect.java (hunk 1 above)
Comment (PILINING) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/AbstractSQLDialect.java]: Hi, @E1izabeth, I have made the change, thanks for the suggestion!

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/AbstractSQLDialect.java [hunk 2]
```diff
@@ -316,7 +316,7 @@ public String[] getInnerBlockPrefixes() {
 
     @Override
     public boolean isWordStart(int ch) {
-        return Character.isUnicodeIdentifierStart(ch);
+    	return Character.isUnicodeIdentifierStart(ch) || ch == '_';
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/AbstractSQLDialect.java]: Please use four spaces instead of tabulation

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/AbstractSQLDialect.java (hunk 2 above)
Comment (PILINING) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/AbstractSQLDialect.java]: Thanks for the reminder, it's been modified.

**Review by E1izabeth — APPROVED** [review]

**Review by ShadelessFox — APPROVED** [review]
👍🏻

**Pull request merged**